backend/internal/service/version_service.go

```diff
@@ -0,0 +1,80 @@
+package service
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync/atomic"
+	"time"
+)
+
+const versionCacheTTL = 15 * time.Minute
+
+type cacheEntry struct {
+	version     string
+	lastFetched time.Time
+}
+
+type VersionService struct {
+	httpClient *http.Client
+	cache      atomic.Value
```

Comment:
Suggest using the generic `atomic.Ptr[*cacheEntry]` which avoids the need to do type casting at runtime